### PR TITLE
Remove set_config when using default client

### PR DESCRIPTION
**Pull request by fjetter**
Symmetric change to
- https://github.com/dask/dask/pull/9808
- https://github.com/dask/dask/pull/9826

Closes https://github.com/dask/distributed/issues/7483
xref https://github.com/dask/dask/issues/9807

### CODE REVIEW

File: continuous_integration/environment-3.10.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ dependencies:
   - zict  # overridden by git tip below
   - zstandard >=0.9.0
   - pip:
-      - git+https://github.com/dask/dask
```
Comment (fjetter) [inline continuous_integration/environment-3.10.yaml:46]: - [ ] Revert to dask/dask@main

**Review by jrbourbeau — COMMENTED** [review]
I went ahead and reverted the temporary environment changes since the corresponding PRs in `dask/dask` have been merged. Overall this looks good to me. The `distributed/tests/test_client.py::test_default_get` test failure does look related though. From skimming the test, I'm inclined to just remove it. We definitely don't need the config checks anymore and the other checks in that that are, I think, covered over in the `dask/dask` test suite (though please correct me if that's not the case)

**Comment by fjetter** [comment]
Indeed, `test_default_get ` is related but can be easily fixed by removing/replacing the config.get checks

The `test_futures_in_subgraphs` is a more interesting case and is actually a possible regression introduced in dask/dask.

The test is submitting a `c.submit(dd.categorical.categorize, ddf, columns=["day"], index=False)` which calculates the result using https://github.com/dask/dask/blob/96a72df0aeb4a7d04844e1ebc2fddb30265a0bc7/dask/dataframe/categorical.py#L152-L154
https://github.com/dask/dask/blob/96a72df0aeb4a7d04844e1ebc2fddb30265a0bc7/dask/base.py#L335-L341
by explicitly proving a `cls` option, a DataFrame. 
https://github.com/dask/dask/pull/9808 changed predecende such that `cls` wins over an implicit default client, see https://github.com/dask/dask/pull/9808#discussion_r1062672836

**Comment by fjetter** [comment]
See for a change in dask/dask to fix the precedence. This is a regression we introduced in the last release https://github.com/dask/dask/pull/9869

**Comment by fjetter** [comment]
I believe the test failures are unrelated. I'll go ahead and merge since this is quite deterministically failing a test in CI. I'm happy to come back and fix anything that's left

**Pull request merged**